frontend/packages/dev-console/src/components/import/import-submit-utils.ts

```diff
@@ -49,12 +50,41 @@ export const createBuildConfig = (
     project: { name: namespace },
     application: { name: application },
     git: { url: repository, ref = 'master', dir: contextDir, secret: secretName },
+    docker: { dockerfilePath },
     image: { tag: selectedTag },
-    build: { env, triggers },
+    build: { env, triggers, strategy: buildStrategy },
     labels: userLabels,
   } = formData;
 
-  const defaultLabels = getAppLabels(name, application, imageStream.metadata.name);
+  const imageStreamName = imageStream && imageStream.metadata.name;
+  const imageStreamNamespace = imageStream && imageStream.metadata.namespace;
+
+  const defaultLabels = getAppLabels(name, application, imageStreamName);
+  let buildStrategyData;
+
+  switch (buildStrategy) {
+    case 'Source':
+      buildStrategyData = {
+        sourceStrategy: {
+          env,
+          from: {
+            kind: 'ImageStreamTag',
+            name: `${imageStreamName}:${selectedTag}`,
+            namespace: imageStreamNamespace,
+          },
+        },
+      };
+      break;
+    case 'Docker':
+      buildStrategyData = {
+        dockerStrategy: { env, dockerfilePath },
+      };
+      break;
+    default:
```

Comment:
Maybe its better to add `Source` strategy i default case so that this logic doesn't break even if there is no valid buildStrategy.

Reply:
Yes, makes sense. Will update it.

Reply:
Done.